ports/rp2/rp2_pio.c

```diff
@@ -487,6 +489,11 @@ STATIC mp_obj_t rp2_state_machine_init_helper(const rp2_state_machine_obj_t *sel
         sm_config_set_in_pins(&config, mp_hal_get_pin_obj(args[ARG_in_base].u_obj));
     }
 
+    // Configure jmp pin, if needed.
+    if (args[ARG_jmp_pin].u_obj != mp_const_none) {
+        sm_config_set_jmp_pin(&config, mp_hal_get_pin_obj(args[ARG_jmp_pin].u_obj));
+    }
+
```

Comment:
would be better if this new block of code was moved down below, between the "set" and "sideset" configuration, to match the order of arguments